save-demo/src/main/kotlin/com/saveourtool/save/demo/service/GithubDownloadToolService.kt

```diff
@@ -0,0 +1,109 @@
+package com.saveourtool.save.demo.service
+
+import com.saveourtool.save.demo.entities.ReleaseAsset
+import com.saveourtool.save.demo.entities.ReleaseMetadata
+import com.saveourtool.save.demo.storage.ToolKey
+import com.saveourtool.save.demo.storage.ToolStorage
+import com.saveourtool.save.demo.utils.toByteBufferFlux
+
+import io.ktor.client.*
```

Comment:
Any particular reason to use ktor here?

Reply:
HttpClient is required here.
Do you suggest using `Spring`'s one?

Reply:
I mean, now that everything is done already, it's not worth changing. But yes, I was asking if there was a particular preference towards `ktor-client` over Spring's `WebClient`

Reply:
The reason was that some code was borrowed from `:save-api` (the REST client), which originally depended on _Ktor_, not _Spring_. For the same reason, `saveourtool/save-backend-tests` depend on _Ktor_.

Reply:
> I mean, now that everything is done already, it's not worth changing. But yes, I was asking if there was a particular preference towards `ktor-client` over Spring's `WebClient`

actually, if we will leave `ktor`, it will allow to use this Service as separate tool (without Spring's ecosystem) to download artifacts from github.